denops/ddc/base/source.ts

```diff
@@ -62,12 +62,10 @@ export abstract class BaseSource {
   }
 
   abstract gatherCandidates(
-    {}: GatherCandidatesArguments,
+    {}: GatherCandidatesArguments<SourceParams>,
   ): Promise<Candidate[]>;
 
-  params(): Record<string, unknown> {
-    return {} as Record<string, unknown>;
-  }
+  abstract params(): SourceParams;
```

Comment:
Why it must abstract?

```typescript
  params(): SourceParams {
    return {} as SourceParams;
  }
```

is invalid?  Please describe it.

Reply:
Oh, I guess we can safely use `as` here, since it will detect if it's wrong. I forget the exact situations, but there are places where `as` can get you into trouble